presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetColumnReader.java

```diff
@@ -169,57 +173,62 @@ public Block readPrimitive(Type type, IntList positions)
         return blockBuilder.build();
     }
 
-    private void readValues(BlockBuilder blockBuilder, int numValues, Type type, IntList positions)
+    private void readValues(BlockBuilder blockBuilder, int numValues, Type type, IntList definitionLevels, IntList repetitionLevels)
```

Comment:
this method looks similar to `skipValues` (the core read logic is the same), can we merge the two somehow so that we get rid of code duplication?

Reply:
agree, refactored